content/blog/scheduling-in-react/index.md

```diff
@@ -0,0 +1,243 @@
+
+```js{1,10-14}
+import { unstable_next } from "scheduler";
+
+function SearchBox(props) {
+  const [inputValue, setInputValue] = React.useState();
+
+  function handleChange(event) {
+    const value = event.target.value;
+
+    setInputValue(value);
+    unstable_next(function() {
+      props.onChange(value);
+      sendAnalyticsNotification(value);
+    });
+  }
+
+  return <input type="text" value={inputValue} onChange={handleChange} />;
+}
+```
+
+The API we’re using, `unstable_next()`, will run the callback with the `Normal` priority. Indeed, our input box already feels a lot more responsive, and frames no longer get dropped while typing into it.
+
+Let’s take another look at the Performance tab together:
+
+![Screenshot of Chrome DevTools that shows that React breaks the rendering work down into small chunks. All frames can be drawn very quickly, although analytics still happen in the middle.](devtools-normal.png)
+
+We see that the long-running tasks are now broken down into smaller ones that can be completed within a single frame. The red triangles that indicate frame drops are also gone.
+
+However, one thing that is still not ideal is that the analytics notification (highlighted in the above screenshot) is still executed with the rendering work. Since the users of our app do not see this task, we can assign it an even lower priority:
+
+```js{3-5,18-22}
+import {
+  unstable_next,
+  unstable_LowPriority,
+  unstable_runWithPriority,
+  unstable_scheduleCallback
+} from "scheduler";
+
+function SearchBox(props) {
+  const [inputValue, setInputValue] = React.useState();
+
+  function handleChange(event) {
+    const value = event.target.value;
+
+    setInputValue(value);
+    unstable_next(function() {
+      props.onChange(value);
+    });
+    unstable_runWithPriority(unstable_LowPriority, function() {
+      unstable_scheduleCallback(function() {
+        sendAnalyticsNotification(value);
+      });
+    });
+  }
+
+  return <input type="text" value={inputValue} onChange={handleChange} />;
+}
+```
+
+If we take another look at the Performance tab with this change and zoom toward the end, we’ll see that our analytics are now sent after all rendering work has completed, and so the total work in our app is perfectly scheduled:
+
+![Screenshot of Chrome DevTools that show that React breaks the rendering work down into small chunks. Analytics are sent at the end after all rendering work has completed.](devtools-normal-and-low.png)
+
+Try it out:
+
+<iframe src="https://codesandbox.io/embed/v0lxm0xlzl?fontsize=14&hidenavigation=1&view=preview" style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+## Limitations of the Scheduler
+
+With the Scheduler, it’s possible to control when certain callbacks can be executed. It is built deep into the latest React implementation and works out of the box with Concurrent mode.
```

Comment:
Can these be joined with "as?" Not sure. I feel like it being built deep into the implementation is a reason why it's possible to control callbacks in the Scheduler, but I'm not so sure that it working out of the box has anyhting to do with this. WDYT?

"With the Scheduler, it's possible to control when certain callbacks can be executed. It is built deep into the latest React implementation and works out of the box with Concurrent mode."

Reply:
I don't think so; The Scheduler is used within React but it is still a separate thing. React itself uses it to schedule certain callbacks but you can schedule callbacks without using React as well.

Reply:
OK. These two sentences feel a bit disjointed, but it's not bad, just something that only I will probably notice. I can't think of anything better so we can leave it as is.

Reply:
I'm ok with that 🙂